### PR TITLE
Support SHOW/DESC NAMESPACE

### PR DESCRIPTION
**Pull request by xuanyuanking**
**Description of changes**

Add `SupportsNamespaces` in UCSingleCatalog class and implemented `listNamespaces` and `loadNamespaceMetadata`, which support `SHOW NAMESPACES` and `DESC NAMESPACE` command.

### CODE REVIEW

**Comment by xuanyuanking** [comment]
cc @cloud-fan

File: connectors/spark/src/main/scala/io/unitycatalog/connectors/spark/UCSingleCatalog.scala [hunk 1]
```diff
@@ -187,4 +213,43 @@ private class UCProxy extends TableCatalog {
...
+    throw new UnsupportedOperationException("Multi-layer namespace is not supported in Unity Catalog")
+  }
+
+  override def loadNamespaceMetadata(namespace: Array[String]): util.Map[String, String] = {
+    checkUnsupportedNestedNamespace(namespace)
+    val schema = try {
+      schemasApi.getSchema(name + "." + namespace(0))
+    } catch {
+      case e: ApiException if e.getCode == 404 =>
+        e.printStackTrace()
```
Comment (cloud-fan) [inline connectors/spark/src/main/scala/io/unitycatalog/connectors/spark/UCSingleCatalog.scala:233]:
```suggestion
```

File: connectors/spark/src/main/scala/io/unitycatalog/connectors/spark/UCSingleCatalog.scala [hunk 2]
```diff
@@ -187,4 +213,43 @@ private class UCProxy extends TableCatalog {
...
+        throw new NoSuchNamespaceException(namespace)
+    }
+    // flatten the schema properties to a map, with the key prefixed by "properties:"
+    val metadata = schema.getProperties.asScala.map {
+        case (k, v) =>  SchemaInfo.JSON_PROPERTY_PROPERTIES + ":" + k -> v
```
Comment (cloud-fan) [inline connectors/spark/src/main/scala/io/unitycatalog/connectors/spark/UCSingleCatalog.scala:238]:
```suggestion
      case (k, v) =>  SchemaInfo.JSON_PROPERTY_PROPERTIES + ":" + k -> v
```

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
also cc @amaliujia

**Pull request merged**